nestkernel/connection_manager.cpp

```diff
@@ -1288,33 +1315,35 @@ nest::ConnectionManager::get_sources( std::vector< index > targets,
 void
 nest::ConnectionManager::get_targets( std::vector< index > sources,
   std::vector< std::vector< index > >& targets,
-  index synapse_model )
+  index synapse_model,
+  std::string post_synaptic_element )
 {
-  thread thread_id;
+  unsigned int thread_id;
   std::vector< index >::iterator source_it;
   std::vector< std::vector< index > >::iterator target_it;
   targets.resize( sources.size() );
   for ( std::vector< std::vector< index > >::iterator i = targets.begin();
         i != targets.end();
-        i++ )
+        ++i )
   {
     ( *i ).clear();
   }
 
-  for ( tVSConnector::iterator it = connections_.begin();
-        it != connections_.end();
-        ++it )
+  // We go through the connections data structure to retrieve all
+  // targets which have an specific post synaptic element for each
+  // source.
+  for ( thread_id = 0; thread_id < connections_.size(); ++thread_id )
   {
-    thread_id = it - connections_.begin();
     // loop over the targets/sources
     source_it = sources.begin();
     target_it = targets.begin();
     for ( ; source_it != sources.end(); source_it++, target_it++ )
     {
-      if ( ( *it ).get( *source_it ) != 0 )
+      if ( validate_source_entry_( thread_id, *source_it ) != 0 )
       {
-        validate_pointer( ( *it ).get( *source_it ) )
-          ->get_target_gids( ( *target_it ), thread_id, synapse_model );
+        validate_pointer( validate_source_entry_( thread_id, *source_it ) )
```

Comment:
Here you call `validate_source_entry_()` for a second time with the same arguments. Perhaps better (guessing that `ConnectorBase` is the right type here
```c++
ConnectorBase const * const source = validate_source_entry_( thread_id, *source_it );
if ( source_ptr != 0 )
{
  source->get_target_gids( *target_it, thread_id, synapse_model );
}
```
Note that I have here assumed that the `validate_pointer()` call is integrated in `validate_source_entry_()`; that would make code here more readable, but if it has side-effects elsewhere, it may mean too much work for now.

Finally, I think the code would become more readable if you changed `target_it` to `targets_it`,  since this is an iterator pointing to a vector.